tests/semantic/test_semantics.py

```diff
@@ -0,0 +1,186 @@
+from pathlib import Path
+
+import pytest
+import pytest_check as check
+from starkware.cairo.lang.vm.vm_exceptions import VmException
+from starkware.starknet.compiler.compile import compile_starknet_files
+from starkware.starknet.testing.state import StarknetState
+from starkware.starkware_utils.error_handling import StarkException
+
+from warp.yul.main import transpile_from_solidity
+from warp.yul.starknet_utils import (
+    deploy_contract,
+    deploy_contract_evm_calldata,
+    invoke_method,
+    invoke_method_evm_calldata,
+)
+
+semantic_tests_folder = os.path.dirname(__file__)
+warp_root = os.path.join(semantic_tests_folder, "../../")
+test_folder = os.path.join(
+    semantic_tests_folder, "solidity/test/libsolidity/semanticTests"
+)
+
+CONTRACT_DEF_REGEX = re.compile(r"contract\s+(\w+)\s+")
+
+
+def get_last_contract(contract_file):
+    contract_name = None
+    with open(contract_file, "r") as contract:
+        lines = contract.readlines()
+        for line in lines:
+            matches = CONTRACT_DEF_REGEX.match(line)
+            if matches:
+                contract_name = matches.groups()[0]
+
+    if contract_name:
+        return contract_name
+
+    pytest.skip(f"Could not find any contract definitions in {contract_file}")
+
+
+def load_ignore_list():
+    with open(
+        os.path.abspath(os.path.join(semantic_tests_folder, "ignore_files.json"))
+    ) as ignore_list:
+        data = json.load(ignore_list)
+
+        if data:
+            return data["files"]
+
+    pytest.skip("Ignore list not found")
+
+
+def load_test_calldata():
+    with open(
+        os.path.abspath(os.path.join(semantic_tests_folder, "test_calldata.json"))
+    ) as calldata:
+        return json.load(calldata)
+
+    pytest.skip("calldata for tests could not be loaded")
+
+
+ignore_list = load_ignore_list()
+test_calldata = load_test_calldata()
+
+
+async def starknet_compile(contract_file):
+    cairo_path = f"{warp_root}/src/warp/cairo-src"
+    return compile_starknet_files(
+        [contract_file], debug_info=True, cairo_path=[cairo_path]
+    )
+
+
+def sol_files_in_dir(folder):
+    # calling str on PosixPath, so that when the test fails it shows the file
+    # name instead of an object
+    return [str(f) for f in Path(folder).glob("**/*.sol")]
+
+
+def split_expectations(expectations):
+    constructor_expectations = []
+    contract_expectations = []
+
+    for e in expectations:
+        constructor_expectations.append(e) if e[
+            "signature"
+        ] == "constructor()" else contract_expectations.append(e)
+
+    if len(constructor_expectations) == 1:
+        return constructor_expectations[0]["callData"], contract_expectations
+
+    return None, contract_expectations
+
+
+def get_expectations(contract_file):
+    split_path = Path(contract_file).parts
+    libsolidity_folder_index = split_path.index("libsolidity")
+    contract_path = f"tests/semantic/solidity/test/{str(Path(*split_path[libsolidity_folder_index:]))}"
```

Comment:
`str(Path)` also feels unnecessary 
![image](https://user-images.githubusercontent.com/44875844/150112648-20ec0162-5c55-4454-9b8a-e309f7c89904.png)

Reply:
In this case its probably not needed I'll try it

Reply:
ack